### ISSUE TITLE
Contact map visualization

### ISSUE THREAD

**Issue by YitianShi** [UVR-WJCHO/HOGraspNet#3]
Hi, I very nice work! I just have some questions regarding the contact map labels. I tried to visualize them by concatenating them with the hand vertices in the same order, but it doesn't gives me the reasonable results at the end. For example, here is the contact (in red) on the hand (in green) which is obviously incorrect. I would really appreciate it if you may correct my implementation of contact visualization.

<img width="1360" height="717" alt="Image" src="https://github.com/user-attachments/assets/429e9dad-93c2-4f64-944b-ccd248bda0be" />

Here is part of my visualization code (inside the dataloader loop): 
```
# Fetch data
hand_scale = sample['mano_scale'][..., None, None]  # Assuming hand_scale is a tensor of shape (N, 1)
hand_xyz_root = sample['mano_xyz_root'].to(device).unsqueeze(1) 
hand_contact = sample['contact']  # reverse the order to match the hand vertices
hand_mano_rot = sample['mano_trans']
hand_mano_pose = sample['mano_pose']
hand_mano_shape = sample['mano_betas']

# Preprocessing
hand_mano_rot = hand_mano_rot.to(device)
hand_mano_pose = hand_mano_pose.to(device)
hand_mano_shape = hand_mano_shape.to(device)
mano_param = torch.cat([hand_mano_rot, hand_mano_pose], dim=-1).to(device)
mano_outcome = mano_layer(global_orient=hand_mano_rot, hand_pose=hand_mano_pose, betas=hand_mano_shape)
mano_verts, mano_joints = mano_outcome.vertices, mano_outcome.joints
mano_verts = mano_verts - mano_joints[:, 0].unsqueeze(1)  # Center the hand at the originq
mano_verts *= 1000.0 / hand_scale.to(device)
mano_joints *= 1000.0 / hand_scale.to(device)
mano_verts = mano_verts.float()    

# Visualize hand contact points with red color, the higher the contact score, the more red
hand_contact_color = torch.zeros((778, 3), device=device)
hand_contact_color[:, 1] = 1.0 - hand_contact[batch]  # Green channel inversely proportional to contact score
hand_contact_color[:, 0] = hand_contact[batch]  # Red channel proportional to contact score
pcd_hand = o3d.geometry.PointCloud()
pcd_hand.points = o3d.utility.Vector3dVector(mano_verts[batch].detach().cpu().numpy())
pcd_hand.colors = o3d.utility.Vector3dVector(hand_contact_color.cpu().numpy())

# Following some visualization with hand pcds ...
```

Thank you very much for your help !!!

**Comment by RbtcsOscar** [comment]
Hi YitianShi.
I have similar problem when loading dataset to 3d point cloud. Did you solve your problem? If you solved, can you share your code?
